### DISCUSSION TITLE
How do re-init lightBox effect after Infinite Scrolling Pagination

### DISCUSSION THREAD

**Discussion by Shadecute** [livewire/livewire#8558] [1. Help] [answered]
I'm using the lightbox feature to display images in a Livewire component. I'm also using Infinite Scrolling to load more images. Everything works fine until I click on the "More" button. The newly loaded images no longer have the lightbox feature, and only the images that were initially loaded still have the lightbox feature. How can I apply this feature to the new images as well??

script in blade component :
```
    @push('scripts')

        <script src="/js/lightbox.js"  crossorigin="anonymous" referrerpolicy="no-referrer" ></script>
        <script>
            document.addEventListener('livewire:initialized', () => {
                const lightbox = new Lightbox();
            })
        </script>

    @endpush
```

images:
```
<div class="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4 gap-4" >
                @foreach($images->slice(6) as $image)
                    <a  href="{{$image->imageAddress}}" class="lightbox transition-all duration-500 group-hover:scale-105" title=" 
                       {{$image->title}}" >
                        <img  src="{{$image->imageAddress}}"  class="rounded-lg" alt="{{$image->title}}" >
                    </a>
                @endforeach
</div>
```
![image info](https://i.ibb.co/tcB6Wnp/Screen-Shot-2024-06-12-at-3-36-13-PM.png)

The lightbox is functioning correctly before click More button for Infinite Scrolling.

![image info](https://i.ibb.co/nQdxzJJ/Screen-Shot-2024-06-12-at-4-02-30-PM.png)

**Comment by PhiloNL** [comment, marked as answer]
You could try the morph hook, this will trigger when new elements have been added:

```
Livewire.hook('morph.added',  ({ el }) => {
   const lightbox = new Lightbox();
})
```

**Reply by Shadecute** [reply]
This was really great. With your guidance, my problem is solved.

**Reply by PhiloNL** [reply]
No problem, happy I could help :)
